shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala

```diff
@@ -90,16 +90,23 @@ object BarChart {
         val numGroups = data.map(_.cluster).distinct.size
         val barsPerGroup = if (numGroups > 1) data.groupBy(_.cluster).map(_._2.size).max else 1
 
+        val fullPlotWidth = xtransformer(plot.xbounds.min + 1.0 * numGroups)
+        val clusterWidth = (fullPlotWidth - numGroups * clusterSpacing) / numGroups
+        val barWidth = (clusterWidth - (barsPerGroup - 1) * spacing) / barsPerGroup
+        val firstClusterPadding = clusterSpacing / 2
+
+        def getClusterStartX(clusterIndex: Int): Double =
+          firstClusterPadding + (clusterWidth + clusterSpacing) * clusterIndex
+        def getBarXInCluster(barIndexInCluster: Int): Double = (barWidth + spacing) * barIndexInCluster
+
         val sorted = data.sortBy(_.cluster)
         val initial: (Double, Drawable) = (sorted.head.cluster, EmptyDrawable())
         sorted.zipWithIndex
           .foldLeft(initial) {
             case ((lastCluster, d), (bar, barIndex)) =>
-              // X offset and bar width.
-              val xscale = 1.0 / barsPerGroup
-              val barx = barIndex * xscale
-              val x = xtransformer(plot.xbounds.min + barx)
-              val barWidth = xtransformer(plot.xbounds.min + barx + xscale) - x
+              // X offset
+              val clusterIndex = if (numGroups > 1) bar.cluster else barIndex
```

Comment:
Idk if this is a reasonable use case, but when I use bar.cluster with just one cluster it ends up defaulting to a regular bar chart, which was a little unexpected.

Reply:
Agreed that that's odd behavior. I'll update so all the bars cluster above the first x label.